### ISSUE TITLE
No Germline variants error

### ISSUE THREAD

**Issue by DanXu-git** [lima1/PureCN#109]
Hello lima,

When I use PureCN with a vcf produced by other variant calling software other than Mutect2,
I got this error below, whether it's because the vcf format is not fit, what critical information or format should the vcf be in order to fit PureCN?

ERROR:
INFO [2019-10-17 09:29:09] Loading coverage files...
WARN [2019-10-17 09:29:09] Coverage data contains single nucleotide intervals.
INFO [2019-10-17 09:29:10] Found log2-ratio in tumor coverage data.
WARN [2019-10-17 09:29:10] Allosome coverage missing, cannot determine sex.
WARN [2019-10-17 09:29:10] Allosome coverage missing, cannot determine sex.
INFO [2019-10-17 09:29:10] Using 41174 intervals (12278 on-target, 28896 off-target).
INFO [2019-10-17 09:29:10] Ratio of mean on-target vs. off-target read counts: NaN
INFO [2019-10-17 09:29:10] Mean off-target bin size: 94490
INFO [2019-10-17 09:29:10] Loading VCF...
INFO [2019-10-17 09:29:10] Found 12753 variants in VCF file.
INFO [2019-10-17 09:29:11] 5852 (45.9%) variants annotated as likely germline (DB INFO flag).
INFO [2019-10-17 09:29:11] Sample1 is tumor in VCF file.
INFO [2019-10-17 09:29:12] No germline variants in VCF.
ERROR: logical subscript contains NAs
STOP

**Comment by lima1** [comment]
Can you share a minimal example VCF to reproduce the error?

**Comment by DanXu-git** [comment]
> Can you share a minimal example VCF to reproduce the error?

It's like this:
chr1	2488450	.	T	C	.	PASS	ADP=274;HET=1;HOM=0;NC=0;WT=0	GT:ABQ:AD:ADF:ADR:DP:FREQ:GQ:PVAL:RBQ:RD:RDF:RDR:SDP	0/1:36:2:1:1:274:0.73%:0:9.8E-1:35:272:115:157:274
chr1	2491184	.	C	T	.	PASS	ADP=197;HET=1;HOM=0;NC=0;WT=0	GT:ABQ:AD:ADF:ADR:DP:FREQ:GQ:PVAL:RBQ:RD:RDF:RDR:SDP	0/1:34:2:2:0:197:1.02%:0:9.8E-1:35:195:149:46:197
chr1	2491205	rs2234161	C	T	.	PASS	ADP=270;DB;HET=1;HOM=0;NC=0;WT=0	GT:ABQ:AD:ADF:ADR:DP:FREQ:GQ:PVAL:RBQ:RD:RDF:RDR:SDP	0/1:35:126:92:34:270:46.67%:0:9.8E-1:34:144:104:40:270
chr1	2491216	.	T	G	.	PASS	ADP=285;HET=1;HOM=0;NC=0;WT=0	GT:ABQ:AD:ADF:ADR:DP:FREQ:GQ:PVAL:RBQ:RD:RDF:RDR:SDP	0/1:34:2:0:2:285:0.7%:0:9.8E-1:34:283:236:47:285
chr1	2491306	rs2234163	G	A	.	PASS	ADP=981;DB;HET=1;HOM=0;NC=0;WT=0	GT:ABQ:AD:ADF:ADR:DP:FREQ:GQ:PVAL:RBQ:RD:RDF:RDR:SDP	0/1:34:420:229:191:981:42.81%:0:9.8E-1:35:561:293:268:981
chr1	2494690	.	C	T	.	PASS	ADP=212;HET=1;HOM=0;NC=0;WT=0	GT:ABQ:AD:ADF:ADR:DP:FREQ:GQ:PVAL:RBQ:RD:RDF:RDR:SDP	0/1:36:4:2:2:212:1.89%:0:9.8E-1:34:208:124:84:212
chr1	2494701	.	C	A	.	PASS	ADP=260;HET=1;HOM=0;NC=0;WT=0	GT:ABQ:AD:ADF:ADR:DP:FREQ:GQ:PVAL:RBQ:RD:RDF:RDR:SDP	0/1:34:2:1:1:260:0.77%:0:9.8E-1:34:258:151:107:260
chr1	2494785	rs8725	G	A	.	PASS	ADP=478;DB;HET=1;HOM=0;NC=0;WT=0	GT:ABQ:AD:ADF:ADR:DP:FREQ:GQ:PVAL:RBQ:RD:RDF:RDR:SDP	0/1:35:253:117:136:478:52.93%:0:9.8E-1:34:225:118:107:478
chr1	2496582	rs2843396	G	C	.	PASS	ADP=1503;DB;HET=0;HOM=1;NC=0;WT=0	GT:ABQ:AD:ADF:ADR:DP:FREQ:GQ:PVAL:RBQ:RD:RDF:RDR:SDP	1/1:34:1503:658:845:1503:100%:0:9.8E-1:0:0:0:0:1503
chr1	3040185	rs2981863	G	C	.	PASS	ADP=82;DB;HET=0;HOM=1;NC=0;WT=0	GT:ABQ:AD:ADF:ADR:DP:FREQ:GQ:PVAL:RBQ:RD:RDF:RDR:SDP	1/1:35:82:42:40:82:100%:0:9.8E-1:0:0:0:0:82

**Comment by DanXu-git** [comment]
> Can you share a minimal example VCF to reproduce the error?

It's like this:
#CHROM  POS     ID      REF     ALT     QUAL    FILTER  INFO    FORMAT  Sample1
chr1	2488450	.	T	C	.	PASS	ADP=274;HET=1;HOM=0;NC=0;WT=0	GT:ABQ:AD:ADF:ADR:DP:FREQ:GQ:PVAL:RBQ:RD:RDF:RDR:SDP	0/1:36:2:1:1:274:0.73%:0:9.8E-1:35:272:115:157:274
chr1	2491184	.	C	T	.	PASS	ADP=197;HET=1;HOM=0;NC=0;WT=0	GT:ABQ:AD:ADF:ADR:DP:FREQ:GQ:PVAL:RBQ:RD:RDF:RDR:SDP	0/1:34:2:2:0:197:1.02%:0:9.8E-1:35:195:149:46:197
chr1	2491205	rs2234161	C	T	.	PASS	ADP=270;DB;HET=1;HOM=0;NC=0;WT=0	GT:ABQ:AD:ADF:ADR:DP:FREQ:GQ:PVAL:RBQ:RD:RDF:RDR:SDP	0/1:35:126:92:34:270:46.67%:0:9.8E-1:34:144:104:40:270
chr1	2491216	.	T	G	.	PASS	ADP=285;HET=1;HOM=0;NC=0;WT=0	GT:ABQ:AD:ADF:ADR:DP:FREQ:GQ:PVAL:RBQ:RD:RDF:RDR:SDP	0/1:34:2:0:2:285:0.7%:0:9.8E-1:34:283:236:47:285
chr1	2491306	rs2234163	G	A	.	PASS	ADP=981;DB;HET=1;HOM=0;NC=0;WT=0	GT:ABQ:AD:ADF:ADR:DP:FREQ:GQ:PVAL:RBQ:RD:RDF:RDR:SDP	0/1:34:420:229:191:981:42.81%:0:9.8E-1:35:561:293:268:981
chr1	2494690	.	C	T	.	PASS	ADP=212;HET=1;HOM=0;NC=0;WT=0	GT:ABQ:AD:ADF:ADR:DP:FREQ:GQ:PVAL:RBQ:RD:RDF:RDR:SDP	0/1:36:4:2:2:212:1.89%:0:9.8E-1:34:208:124:84:212
chr1	2494701	.	C	A	.	PASS	ADP=260;HET=1;HOM=0;NC=0;WT=0	GT:ABQ:AD:ADF:ADR:DP:FREQ:GQ:PVAL:RBQ:RD:RDF:RDR:SDP	0/1:34:2:1:1:260:0.77%:0:9.8E-1:34:258:151:107:260
chr1	2494785	rs8725	G	A	.	PASS	ADP=478;DB;HET=1;HOM=0;NC=0;WT=0	GT:ABQ:AD:ADF:ADR:DP:FREQ:GQ:PVAL:RBQ:RD:RDF:RDR:SDP	0/1:35:253:117:136:478:52.93%:0:9.8E-1:34:225:118:107:478
chr1	2496582	rs2843396	G	C	.	PASS	ADP=1503;DB;HET=0;HOM=1;NC=0;WT=0	GT:ABQ:AD:ADF:ADR:DP:FREQ:GQ:PVAL:RBQ:RD:RDF:RDR:SDP	1/1:34:1503:658:845:1503:100%:0:9.8E-1:0:0:0:0:1503
chr1	3040185	rs2981863	G	C	.	PASS	ADP=82;DB;HET=0;HOM=1;NC=0;WT=0	GT:ABQ:AD:ADF:ADR:DP:FREQ:GQ:PVAL:RBQ:RD:RDF:RDR:SDP	1/1:35:82:42:40:82:100%:0:9.8E-1:0:0:0:0:82

**Comment by lima1** [comment]
Can you share the header as well? Or which variant caller generated this file?

**Comment by DanXu-git** [comment]
> Can you share the header as well? Or which variant caller generated this file?

It was generated by VarScan.
And the header is:

##fileformat=VCFv4.2
##FILTER=<ID=indelError,Description="Likely artifact due to indel reads at this position">
##FILTER=<ID=str10,Description="Less than 10% or more than 90% of variant supporting reads on one strand">
##FORMAT=<ID=ABQ,Number=1,Type=Integer,Description="Average quality of variant-supporting bases (qual2)">
##FORMAT=<ID=AD,Number=R,Type=Integer,Description="Allelic depths for the ref and alt alleles in the order listed">
##FORMAT=<ID=ADF,Number=1,Type=Integer,Description="Depth of variant-supporting bases on forward strand (reads2plus)">
##FORMAT=<ID=ADR,Number=1,Type=Integer,Description="Depth of variant-supporting bases on reverse strand (reads2minus)">
##FORMAT=<ID=DP,Number=1,Type=Integer,Description="Quality Read Depth of bases with Phred score >= 0">
##FORMAT=<ID=FREQ,Number=1,Type=String,Description="Variant allele frequency">
##FORMAT=<ID=GQ,Number=1,Type=Integer,Description="Genotype Quality">
##FORMAT=<ID=GT,Number=1,Type=String,Description="Genotype">
##FORMAT=<ID=PVAL,Number=1,Type=String,Description="P-value from Fisher's Exact Test">
##FORMAT=<ID=RBQ,Number=1,Type=Integer,Description="Average quality of reference-supporting bases (qual1)">
##FORMAT=<ID=RD,Number=1,Type=Integer,Description="Depth of reference-supporting bases (reads1)">
##FORMAT=<ID=RDF,Number=1,Type=Integer,Description="Depth of reference-supporting bases on forward strand (reads1plus)">
##FORMAT=<ID=RDR,Number=1,Type=Integer,Description="Depth of reference-supporting bases on reverse strand (reads1minus)">
##FORMAT=<ID=SDP,Number=1,Type=Integer,Description="Raw Read Depth as reported by SAMtools">
##INFO=<ID=ADP,Number=1,Type=Integer,Description="Average per-sample depth of bases with Phred score >= 0">
##INFO=<ID=DB,Number=0,Type=Flag,Description="dbSNP Membership">
##INFO=<ID=HET,Number=1,Type=Integer,Description="Number of samples called heterozygous-variant">
##INFO=<ID=HOM,Number=1,Type=Integer,Description="Number of samples called homozygous-variant">
##INFO=<ID=NC,Number=1,Type=Integer,Description="Number of samples not called">
##INFO=<ID=WT,Number=1,Type=Integer,Description="Number of samples called reference (wild-type)">
##contig=<ID=chrM,length=16571>
##contig=<ID=chr1,length=249250621>
##contig=<ID=chr2,length=243199373>
##contig=<ID=chr3,length=198022430>
##contig=<ID=chr4,length=191154276>
##contig=<ID=chr5,length=180915260>
##contig=<ID=chr6,length=171115067>
##contig=<ID=chr7,length=159138663>
##contig=<ID=chr8,length=146364022>
##contig=<ID=chr9,length=141213431>
##contig=<ID=chr10,length=135534747>
##contig=<ID=chr11,length=135006516>
##contig=<ID=chr12,length=133851895>
##contig=<ID=chr13,length=115169878>
##contig=<ID=chr14,length=107349540>
##contig=<ID=chr15,length=102531392>
##contig=<ID=chr16,length=90354753>
##contig=<ID=chr17,length=81195210>
##contig=<ID=chr18,length=78077248>
##contig=<ID=chr19,length=59128983>
##contig=<ID=chr20,length=63025520>
##contig=<ID=chr21,length=48129895>
##contig=<ID=chr22,length=51304566>
##contig=<ID=chrX,length=155270560>
##contig=<ID=chrY,length=59373566>
##contig=<ID=chr1_gl000191_random,length=106433>
##contig=<ID=chr1_gl000192_random,length=547496>
##contig=<ID=chr4_ctg9_hap1,length=590426>
##contig=<ID=chr4_gl000193_random,length=189789>
##contig=<ID=chr4_gl000194_random,length=191469>
##contig=<ID=chr6_apd_hap1,length=4622290>
##contig=<ID=chr6_cox_hap2,length=4795371>
##contig=<ID=chr6_dbb_hap3,length=4610396>
##contig=<ID=chr6_mann_hap4,length=4683263>
##contig=<ID=chr6_mcf_hap5,length=4833398>
##contig=<ID=chr6_qbl_hap6,length=4611984>
##contig=<ID=chr6_ssto_hap7,length=4928567>
##contig=<ID=chr7_gl000195_random,length=182896>
##contig=<ID=chr8_gl000196_random,length=38914>
##contig=<ID=chr8_gl000197_random,length=37175>
##contig=<ID=chr9_gl000198_random,length=90085>
##contig=<ID=chr9_gl000199_random,length=169874>
##contig=<ID=chr9_gl000200_random,length=187035>
##contig=<ID=chr9_gl000201_random,length=36148>
##contig=<ID=chr11_gl000202_random,length=40103>
##contig=<ID=chr17_ctg5_hap1,length=1680828>
##contig=<ID=chr17_gl000203_random,length=37498>
##contig=<ID=chr17_gl000204_random,length=81310>
##contig=<ID=chr17_gl000205_random,length=174588>
##contig=<ID=chr17_gl000206_random,length=41001>
##contig=<ID=chr18_gl000207_random,length=4262>
##contig=<ID=chr19_gl000208_random,length=92689>
##contig=<ID=chr19_gl000209_random,length=159169>
##contig=<ID=chr21_gl000210_random,length=27682>
##contig=<ID=chrUn_gl000211,length=166566>
##contig=<ID=chrUn_gl000212,length=186858>
##contig=<ID=chrUn_gl000213,length=164239>
##contig=<ID=chrUn_gl000214,length=137718>
##contig=<ID=chrUn_gl000215,length=172545>
##contig=<ID=chrUn_gl000216,length=172294>
##contig=<ID=chrUn_gl000217,length=172149>
##contig=<ID=chrUn_gl000218,length=161147>
##contig=<ID=chrUn_gl000219,length=179198>
##contig=<ID=chrUn_gl000220,length=161802>
##contig=<ID=chrUn_gl000221,length=155397>
##contig=<ID=chrUn_gl000222,length=186861>
##contig=<ID=chrUn_gl000223,length=180455>
##contig=<ID=chrUn_gl000224,length=179693>
##contig=<ID=chrUn_gl000225,length=211173>
##contig=<ID=chrUn_gl000226,length=15008>
##contig=<ID=chrUn_gl000227,length=128374>
##contig=<ID=chrUn_gl000228,length=129120>
##contig=<ID=chrUn_gl000229,length=19913>
##contig=<ID=chrUn_gl000230,length=43691>
##contig=<ID=chrUn_gl000231,length=27386>
##contig=<ID=chrUn_gl000232,length=40652>
##contig=<ID=chrUn_gl000233,length=45941>
##contig=<ID=chrUn_gl000234,length=40531>
##contig=<ID=chrUn_gl000235,length=34474>
##contig=<ID=chrUn_gl000236,length=41934>
##contig=<ID=chrUn_gl000237,length=45867>
##contig=<ID=chrUn_gl000238,length=39939>
##contig=<ID=chrUn_gl000239,length=33824>
##contig=<ID=chrUn_gl000240,length=41933>
##contig=<ID=chrUn_gl000241,length=42152>
##contig=<ID=chrUn_gl000242,length=43523>
##contig=<ID=chrUn_gl000243,length=43341>
##contig=<ID=chrUn_gl000244,length=39929>
##contig=<ID=chrUn_gl000245,length=36651>
##contig=<ID=chrUn_gl000246,length=38154>
##contig=<ID=chrUn_gl000247,length=36422>
##contig=<ID=chrUn_gl000248,length=39786>
##contig=<ID=chrUn_gl000249,length=38502>
##source=VarScan2
#CHROM	POS	ID	REF	ALT	QUAL	FILTER	INFO	FORMAT	Sample1

**Comment by lima1** [comment]
Thanks a lot, I'll try to run Varscan2 on some of my test samples. In general it should work, should be a minor fix. Might take a few days though...

**Comment by lima1** [comment]
I couldn't test Varscan2 due to their license, but the VCF you provided has an invalid AD field where the reference count is missing. I added code to fill it and it seems to work. Let me know if this fixes the issue.

**Comment by DanXu-git** [comment]
I've tried a bunch of samples data, it totally worked, thanks a lot !

If it is ok that I use only the snp information but the indel results for segmentation and purity, ploidy detection?

**Comment by lima1** [comment]
Yes, indels are tricky anyways because their estimated allelic fractions can be fairly inaccurate, depending on the caller.

Make sure to provide a mapping bias RDS file and for non-Mutect callers, do enhanced artifact filtering (things like strand-bias) upstream of PureCN. But you will notice when artifacts are a problem, see the main vignette.

**Comment by DanXu-git** [comment]
Okay, thank you, I'll look into it.

**Comment by DanXu-git** [comment]
Dear lima,

When I generate mapping bias RDS file, which was successfully generated, I found that the interval_weights.txt file didn't show up, so I wonder if it's the same problem given that I use the varscan2 merged vcf.  Consider that when I used mutect vcfs it worked well.

There's no error in log, but it looks like it stopped suddenly(end of log):
WARN [2019-10-22 11:34:42] Allosome coverage missing, cannot determine sex.
WARN [2019-10-22 11:34:42] Allosome coverage missing, cannot determine sex.
WARN [2019-10-22 11:34:42] Allosome coverage missing, cannot determine sex.
WARN [2019-10-22 11:34:42] Allosome coverage missing, cannot determine sex.
WARN [2019-10-22 11:34:42] Allosome coverage missing, cannot determine sex.
WARN [2019-10-22 11:34:42] Allosome coverage missing, cannot determine sex.
INFO [2019-10-22 11:34:42] Processing on-target regions...
INFO [2019-10-22 11:34:43] Removing 1344 intervals with low coverage in normalDB.
INFO [2019-10-22 11:34:43] Removing 8 intervals with zero coverage in more than 3% of normalDB.

my Script is like:
Rscript NormalDB.R --outdir ./ --coveragefiles loess_normal.list --genome hg19 --assay VarScan --normal_panel VarScan2.anno.vcf.gz

It supposed to have these two files I think, which .png file showed up but the .txt file:
interval_weights.png
interval_weights.txt
